src/image_gen/http_image_gen_calculator.cc

```diff
@@ -104,25 +127,55 @@ class ImageGenCalculator : public CalculatorBase {
         auto pipe = it->second;
 
         auto payload = cc->Inputs().Tag(INPUT_TAG_NAME).Get<ovms::HttpPayload>();
-        if (payload.parsedJson->HasParseError())
-            return absl::InvalidArgumentError("Failed to parse JSON");
 
-        if (!payload.parsedJson->IsObject()) {
-            return absl::InvalidArgumentError("JSON body must be an object");
+        std::unique_ptr<ov::Tensor> images;  // output
+
+        if (absl::StartsWith(payload.uri, "/v3/images/generations")) {
+            if (payload.parsedJson->HasParseError())
+                return absl::InvalidArgumentError("Failed to parse JSON");
+
+            if (!payload.parsedJson->IsObject()) {
+                return absl::InvalidArgumentError("JSON body must be an object");
+            }
+
+            SET_OR_RETURN(std::string, prompt, getPromptField(*payload.parsedJson));
+            SET_OR_RETURN(ov::AnyMap, requestOptions, getImageGenerationRequestOptions(*payload.parsedJson, pipe->args));
+
+            ov::genai::Text2ImagePipeline request = pipe->text2ImagePipeline->clone();
+
+            auto status = generateTensor(request, prompt, requestOptions, images);
+            if (!status.ok()) {
+                return status;
+            }
+        } else if (absl::StartsWith(payload.uri, "/v3/images/edits")) {
+            if (payload.multipartParser->hasParseError())
+                return absl::InvalidArgumentError("Failed to parse multipart data");
+
```

Comment:
Is multipartParser->hasParseError() check enough? Don't we need also parsedJson checks as for text2img?

Reply:
no, we dont use parsedJson at all, and this in fact will always have parsing error - because there is no json in the /edit request